vms/platformvm/txs/mempool/mempool.go

```diff
@@ -224,26 +233,33 @@ func (m *mempool) HasTxs() bool {
 }
 
 func (m *mempool) PeekTxs(maxTxsBytes int) []*txs.Tx {
-	txs := m.unissuedDecisionTxs.List()
-	txs = append(txs, m.unissuedStakerTxs.List()...)
-
-	size := 0
-	for i, tx := range txs {
-		size += len(tx.Bytes())
-		if size > maxTxsBytes {
-			return txs[:i]
-		}
+	txs := make([]*txs.Tx, 0)
+	txsSizeSum := 0
+
+	txIter := m.unissuedDecisionTxs.NewIterator()
+	for txsSizeSum <= maxTxsBytes && txIter.Next() {
+		tx := txIter.Value()
+		txs = append(txs, tx)
+		txsSizeSum += tx.Size()
```

Comment:
This logic doesn't look right anymore. Previously we ensured that the sum of all the tx sizes returned by this function would be `<= maxTxsBytes`. I don't think this is still the case.

Reply:
Seems like this would be a good regression test to add ^

Reply:
Just fixed with a unit test.